### PR TITLE
dev-db/sqliteman: Depend on x11-libs/qscintilla[qt5]

### PR DESCRIPTION
**Pull request by jok-ts**
Gentoo-Bug: 634982

Patch is modelled after the rough Qt5 port in [Valeriy Malov's upstream PR 22](https://github.com/pvanek/sqliteman/pull/22).

### CODE REVIEW

File: dev-db/sqliteman/files/sqliteman-1.2.2-qt5.patch [hunk 1]
```diff
@@ -0,0 +1,346 @@
...
+index d9fd041..f589bfe 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -101,28 +101,10 @@ 
+ 
+ 
+ SET (QT_MT_REQUIRED true)
+-SET (QT_MIN_VERSION "4.2.0")
++SET (QT_MIN_VERSION "5.7.1")
+ SET( QT_USE_QTSQL TRUE )
```
Comment (a17r) [inline dev-db/sqliteman/files/sqliteman-1.2.2-qt5.patch]: You can drop these QT_USE_* vars.

File: dev-db/sqliteman/files/sqliteman-1.2.2-qt5.patch [hunk 2]
```diff
@@ -0,0 +1,346 @@
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index d9fd041..f589bfe 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -101,28 +101,10 @@ 
+ 
+ 
+ SET (QT_MT_REQUIRED true)
```
Comment (a17r) [inline dev-db/sqliteman/files/sqliteman-1.2.2-qt5.patch]: This has no effect on Qt5 anymore (always multithreaded)

File: dev-db/sqliteman/sqliteman-1.2.2-r4.ebuild [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+LICENSE="GPL-2"
+SLOT="0"
+KEYWORDS="~amd64 ~arm ~x86"
+IUSE=""
+
+RDEPEND="
+	dev-qt/qtcore:5
+	dev-qt/qtgui:5
+	dev-qt/qtsql:5[sqlite]
```
Comment (a17r) [inline dev-db/sqliteman/sqliteman-1.2.2-r4.ebuild]: This is likely missing a `qtwidgets:5` dependency, if our cmake code ain't wrong

File: dev-db/sqliteman/files/sqliteman-1.2.2-qt5.patch [hunk 4]
```diff
@@ -0,0 +1,346 @@
...
+ 
+-    SET(QSCINTILLA_NAMES ${QSCINTILLA_NAMES} qscintilla2 libqscintilla2)
+-    FIND_LIBRARY(QSCINTILLA_LIBRARY
+-        NAMES ${QSCINTILLA_NAMES}
+-        PATHS ${QT_LIBRARY_DIR}
+-    )
++    SET(QSCINTILLA_NAMES ${QSCINTILLA_NAMES} qscintilla2 libqscintilla2 qscintilla2_qt5 libqscintilla2_qt5)
++    FIND_LIBRARY(QSCINTILLA_LIBRARY
++        NAMES ${QSCINTILLA_NAMES}
++        HINTS /usr/lib64/qt5
```
[the remark this answers is not in the document]
Comment (jok-ts) [reply dev-db/sqliteman/files/sqliteman-1.2.2-qt5.patch]: I'll be damned, that works. I'm almost sure it didn't when I tried that on an other package a month ago.

**Comment by jok-ts** [comment]
I believe I got all the requested changes in and the package _still_ builds and works.

**Comment by a17r** [comment]
@kensington, what do you think?

**Comment by kensington** [comment]
If it works, ship it!

**Comment by jok-ts** [comment]
@a17r , should I make that change and rebase?

**Comment by a17r** [comment]
yes please! and then we'll merge asap.

**Pull request closed without merging**